pulpcore/app/models/repository.py

```diff
@@ -474,3 +507,52 @@ def __exit__(self, exc_type, exc_value, traceback):
         else:
             self.complete = True
             self.save()
+            self.compute_counts()
+
+
+class RepositoryVersionContentDetails(models.Model):
+    ADDED = 'A'
+    PRESENT = 'P'
+    REMOVED = 'R'
+    COUNT_TYPE_CHOICES = (
+        (ADDED, 'added'),
+        (PRESENT, 'present'),
+        (REMOVED, 'removed'),
+    )
+
+    count_type = models.CharField(
+        max_length=1,
+        choices=COUNT_TYPE_CHOICES,
+    )
+    content_type = models.TextField()
+    repository_version = models.ForeignKey('RepositoryVersion', related_name='counts',
+                                           on_delete=models.CASCADE)
+    count = models.IntegerField()
+
+    @property
+    def content_href(self):
+        """
+        Generate URLs for the content types present in the RepositoryVersion.
+
+        For each content type present in the RepositoryVersion, create the URL of the viewset of
+        that variety of content along with a query parameter which filters it by presence in this
+        RepositoryVersion.
+
+        Args:
+            obj (pulpcore.app.models.RepositoryVersion): The RepositoryVersion being serialized.
+        Returns:
+            dict: {<_type>: <url>}
+        """
+        ctype_model = Content.objects.filter(_type=self.content_type).first().cast().__class__
+        ctype_view = get_view_name_for_model(ctype_model, 'list')
+        try:
+            ctype_url = reverse(ctype_view)
+        except django.urls.exceptions.NoReverseMatch:
+            # We've hit a content type for which there is no viewset.
+            # There's nothing we can do here, except to skip it.
+            return
+        rv_href = "/pulp/api/v3/repositories/{repo}/versions/{version}/".format(
+            repo=self.repository_version.repository.pk, version=self.repository_version.number)
+        full_url = "{base}?repository_version={rv_href}".format(
```

Comment:
This string needs to be toggled on the type of the count, e.g. if it's the href for content being added, it should be ```"{base}?repository_version_added={rv_href}"``` and if it's removed it should be ```"{base}?repository_version_removed={rv_href}"```.   For "present" it would stay as it is currently.

Reply:
fixed in this push. Thanks @dralley !